include/crocoddyl/core/actions/lqr.hxx

```diff
@@ -44,7 +44,7 @@ void ActionModelLQRTpl<Scalar>::calc(const boost::shared_ptr<ActionDataAbstract>
   } else {
     data->xnext.noalias() = Fx_ * x + Fu_ * u + f0_;
   }
-  data->cost = 0.5 * x.dot(Lxx_ * x) + 0.5 * u.dot(Luu_ * u) + x.dot(Lxu_ * u) + lx_.dot(x) + lu_.dot(u);
+  data->cost = Scalar(0.5) * x.dot(Lxx_ * x) + 0.5 * u.dot(Luu_ * u) + x.dot(Lxu_ * u) + lx_.dot(x) + lu_.dot(u);
```

Comment:
You missed the scalar casting here.

Reply:
Nice catch! I will create another PR tomorrow with all your annotations :)

Reply:
Solved this issue and other in diff-lqr in my ongoing PR.